pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -104,10 +108,25 @@ func (pl *VolumeZone) Filter(ctx context.Context, _ *framework.CycleState, pod *
 
 	for i := range pod.Spec.Volumes {
 		volume := pod.Spec.Volumes[i]
-		if volume.PersistentVolumeClaim == nil {
+		pvcName := ""
+		ephemeral := false
+		switch {
+		case volume.PersistentVolumeClaim != nil:
```

Comment:
Is there a utility function we can make for this? It seems like this same logic is used in many places now.

Reply:
We discussed that when originally adding generic ephemeral volumes. The conclusion at that time was that there is no good place for such an utility function. It would be very small, too, so changing dependencies just for it might not be justified.

Do you have a suggestion where it could be place such that kube-controller-manager, kube-apiserver and kubelet can call it?

Reply:
We have used pkg/volume/util in the past to put methods shared by scheduler and kubelet

Reply:
I have started defining and using such helper methods in https://github.com/pohly/kubernetes/commits/generic-ephemeral-volume-helpers

I'll also convert all other code, to be sure that the proposed function signature works in all cases. Then we can start the bike shedding about the function names :sweat_smile: 

But let's do that in a separate PR once this here is merged. I had to rebase once more because of the changes in `registry.go`.